src/healthCheck/BaseHealthCheck.js

```diff
@@ -0,0 +1,157 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * Lewis Daly <lewis@vesselstech.com>
+ --------------
+ ******/
+
+'use strict'
+
+const {
+  statusEnum
+} = require('./HealthCheckEnums')
+const Logger = require('../logger')
+
+/**
+ * @class BaseHealthCheck
+ *
+ *
+ * BaseHealthCheck is a common class for handling health checks across multiple projects. If you wish to test
+ * subservices, extend BaseHealthCheck and implement the required methods.
+ *
+ * @example
+ *  // Extend the BaseHealthCheck class, and implement the required getSubServiceHealth<serviceName> methods.
+ *
+ *  class HealthCheck extends BaseHealthCheck {
+ *
+ *     async getSubServiceHealthDatastore() {
+ *       // implement for this service
+ *     }
+ *
+ *     async getSubServiceHealthBroker() {
+ *       // implement for this service
+ *     }
+ *   }
+ *
+ *
+ *
+ */
+class BaseHealthCheck {
+  constructor (packageJson, servicesToCheck) {
+    this.packageJson = packageJson
+    this.servicesToCheck = servicesToCheck
+
+    this.getHealth = this.getHealth.bind(this)
+    this.getSubServiceHealth = this.getSubServiceHealth.bind(this)
+  }
+
+  /**
+   * @function getHealth
+   *
+   * @description Gets the health of the service along with sub-services
+   *
+   */
+  async getHealth () {
+    // Default values
+    let status = statusEnum.OK
+    let isHealthy = true
+    let subServices = {}
+
+    const uptime = process.uptime() // in seconds by default
```

Comment:
The uptime, startTime etc will report on the server that is started in the common lib and not the requested API.
I believe that this may need to be implemented in each API or have an override function in HealthCheck to populate these and add it to the response object

Reply:
I'm not sure about that. If we import the Health check server and start it along with the rest of the service, then the `process.uptime()` should be the same. The common lib won't be started as a different service... right?

Here's an example implementation if that helps. https://github.com/vessels-tech/email-notifier/blob/feature/796-improve-health-checks/src/setup.js#L50